pkg/oauthserver/authenticator/password/keystonepassword/keystonepassword.go

```diff
@@ -16,24 +18,116 @@ import (
 	authapi "github.com/openshift/origin/pkg/oauthserver/api"
 )
 
+type keystoneDomain struct {
+	ID    string  `json:string`
+	Name  string  `json:string`
+}
+
+type keystoneUser struct {
+	ID                 string          `json:string`
+	Name               string          `json:string`
+	Domain             keystoneDomain  `json:domain`
+	PasswordExpiresAt  *string         `json:password_expires_at`
+}
+
+type keystoneToken struct {
+	Methods    []string      `json:methods`
```

Comment:
you can omit all unused fields and levels

Reply:
Done